pkg/storage/postgres/filter.go

```diff
@@ -39,8 +40,12 @@ func addFilterExpressionToQuery(query *string, args *[]interface{}, expr storage
 			*args = append(*args, expr.Value)
 			return nil
 		case "$index":
-			*query += schemaName + "." + recordsTableName + ".index_cidr >>= " + fmt.Sprintf("$%d", len(*args)+1)
-			*args = append(*args, expr.Value)
+			if isCIDR(expr.Value) {
+				*query += schemaName + "." + recordsTableName + ".index_cidr >>= " + fmt.Sprintf("$%d", len(*args)+1)
+				*args = append(*args, expr.Value)
+			} else {
+				*query += " false "
```

Comment:
why do we set query to `false` and not return an error instead?

Reply:
It's not technically an error. Like you may want to search on an indexed field that's not a CIDR. We don't currently support other types of indexes, but might in the future. We use `false` because we have to put something otherwise postgres is going to complain about an invalid query, but we actually just don't want to perform the search on that field. (in other words if I return an error here it will still be broken because we always do a search by id and index, and we can't always control the input to make sure its valid)

Reply:
I'm confused - per RFC the index was strongly typed

```json
"$index": { "cidr": "192.168.0.0/16" }
```